minio/resource_minio_kms_key.go

```diff
@@ -0,0 +1,94 @@
+package minio
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMinioKMSKey() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: minioCreateKMSKey,
+		ReadContext:   minioReadKMSKey,
+		DeleteContext: minioDeleteKMSKey,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func minioCreateKMSKey(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	keyConfig := KMSKeyConfig(d, meta)
+
+	keyID := keyConfig.MinioKMSKeyID
+
+	if err := keyConfig.MinioAdmin.CreateKey(ctx, keyID); err != nil {
+		return NewResourceError("error creating service account", keyID, err)
+	}
+
+	d.SetId(aws.StringValue(&keyID))
+	d.Set("key_id", d.Id())
```

Comment:
Suggestion for fixing the `G104 (CWE-703): Errors unhandled. (Confidence: HIGH, Severity: LOW)` error pointed by `gosec`:
```suggestion
	_ = d.Set("key_id", d.Id())
```

Reply:
Thank you, I fixed it accordingly.